src/App.js

```diff
@@ -1,23 +1,27 @@
-import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
+import { observer } from 'mobx-react';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
 
-import Container from '@containers/Container'
-import Splash from '@containers/Splash'
+import PrivateRoute from '@components/PrivateRoute';
+import Landing from '@containers/Landing'
 import Uploads from '@containers/Uploads'
+import Store from '@models/Store';
 
 import './App.css';
 
-function App() {
+const App = observer(() => {
   return (
-      <main className="d-flex h-100 flex-column">
-        <BrowserRouter forceRefresh={false}>
-          <Switch>
-            <Route path='/' component={Splash} exact />
-            <Route path='/success' component={Container} exact />
-            <Route path='/uploads' component={Uploads} exact />
-          </Switch>
-        </BrowserRouter>
-      </main>
+    <main className="d-flex h-100 flex-column">
+      <BrowserRouter forceRefresh={false}>
+        <Switch>
+          <Route path='/' component={Landing} exact />
+          <Route path='/success' component={Landing} exact />
+          <PrivateRoute path="/uploads" isAllowed={Store.user}>
+            <Uploads />
+          </PrivateRoute>
+        </Switch>
+      </BrowserRouter>
+    </main>
```

Comment:
why the xml encoding here? Not an option to do this "in code" or package it as xml data to be loaded in? Too slow to do that maybe?

Reply:
no xml. This is all HTML. So this is a bit trickier: 

`<main>` that's just html main tag.
`<BrowserRouter>` this is a package you can see at the top but basically tells react how to define routes.
`<Route>` checks which "route" is the user in, and render the component allocated to that route.

You'll get more familiar with those concepts

Reply:
technically html is a subset of xml :P